### PR TITLE
feat: Add new regexp filter for specific tests within suites (-exclude_discrete_tests)

### PR DESCRIPTION
**Pull request by kajiya3141**
Add the ability to skip/not parse the results of any individual tests within the different suites. Uses regexp and flag.Lookup.

### CODE REVIEW

**Comment by kajiya3141** [comment]
Hey everyone!

Thanks for looking at my PR 😄 I took Amy's advice and made the filter more generic/top-level. Have a look and let me know if it looks good :sunglasses: Thanks in advance!

**Comment by a-crate** [comment]
I like this approach, I think there is a way for us to handle this flag more directly though. I can't do a full review at the moment but I will get back to you by end of week with one.

File: cmd/manager/main.go
Comment (a-crate) [inline cmd/manager/main.go:166]: You can drop the else here since the other call is log.Fatal

File: testworkflow.go
Comment (a-crate) [inline testworkflow.go:1056]: I like the flag, but there is a more direct way to skip these tests than modifying test results (this method would also create some inconsistencies in exit status from CIT and XML results content).

The test binary should be executed using the flag `-test.skip` and the value of `exclude_discrete_tests`. All of our current logic uses positive matching to select test cases so this is a good use of the negative match flag. It's a bit convoluted to pass the flag all the way there, but you can  add the `exclude_discrete_tests` string representation to the `TestWorkflow struct`, pass the value to the VM in `setInstanceTestMetadata` and then fetch it in `wrapper/main.go` and add it to `testArguments`.

**Review by a-crate — APPROVED** [review]

**Comment by a-crate** [comment]
Sorry, wrong button, didn't mean to click approve. It won't go through until I click approve in an internal tool anyway.

**Comment by kajiya3141** [comment]
Hey @a-crate! Thanks for the review and the comments - I'll look to implement the changes now we're back from the engineering sprint :)

**Comment by kajiya3141** [comment]
@a-crate Ready when you are :smile:

File: testworkflow.go
Comment (kajiya3141) [reply testworkflow.go]: All done :saluting_face:

File: testworkflow.go
Comment (a-crate) [inline testworkflow.go:708]: Can you add this to the function signature so that the library does not have a dependency on the manager? Otherwise LGTM and ready to merger

Comment (kajiya3141) [reply testworkflow.go]: Sorry @a-crate, I got a bit sidetracked :smile: Is this what you meant?

Comment (a-crate) [reply testworkflow.go]: Yep, thanks

**Comment by a-crate** [comment]
You might need to update the unit test for that function too, in `testworkflow_test.go` line 596. Sorry, forgot to enable tests earlier.

**Comment by kajiya3141** [comment]
> You might need to update the unit test for that function too, in `testworkflow_test.go` line 596. Sorry, forgot to enable tests earlier.

No worries, leave it with me and I'll push as a separate commit :smile:

**Comment by a-crate** [comment]
Can you register your github email (`79866126+kajiya3141​@users.noreply.github.com`) as an email for the CLA too? That's causing some CLA checks to fail

**Comment by kajiya3141** [comment]
> Can you register your github email (`79866126+kajiya3141​@users.noreply.github.com`) as an email for the CLA too? That's causing some CLA checks to fail

That's weird, I thought I had my email set to public on here :thinking: No worries I'll add it to the CLA now :smile:

**Pull request merged**
